# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mozglue/baseprofiler/core/ProfilerBacktrace.h
# Commit: fd3a29881783
# Full Hash: fd3a29881783aad22258c96c6ea04220a40f2648
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2019-09-18 09:56:40
# Regressor Bug: 1576551
# File Overlap Count: 1
# Description:
#   Bug 1576551 - Store BlocksRingBuffer outside of ProfileBuffer - r=gregtatum
#   
#   The main `BlocksRingBuffer`s will be stored in `CorePS` (outside of
#   `ProfileBuffer`s), as we need to be able to safely access the underlying buffers
#   when profilers are not enabled.
# ==============================================================================

diff -r 3dfc76ea7bd9 -r fd3a29881783 mozglue/baseprofiler/core/ProfilerBacktrace.h
--- a/mozglue/baseprofiler/core/ProfilerBacktrace.h	Wed Sep 18 01:19:14 2019 +0000
+++ b/mozglue/baseprofiler/core/ProfilerBacktrace.h	Wed Sep 18 01:19:21 2019 +0000
@@ -11,6 +11,7 @@
 
 namespace mozilla {
 
+class BlocksRingBuffer;
 class TimeStamp;
 
 namespace baseprofiler {
@@ -24,7 +25,8 @@
 class ProfilerBacktrace {
  public:
   ProfilerBacktrace(const char* aName, int aThreadId,
-                    UniquePtr<ProfileBuffer> aBuffer);
+                    UniquePtr<BlocksRingBuffer> aBlocksRingBuffer,
+                    UniquePtr<ProfileBuffer> aProfileBuffer);
   ~ProfilerBacktrace();
 
   // ProfilerBacktraces' stacks are deduplicated in the context of the
@@ -43,7 +45,10 @@
 
   UniqueFreePtr<char> mName;
   int mThreadId;
-  UniquePtr<ProfileBuffer> mBuffer;
+  // `BlocksRingBuffer` in which `mProfileBuffer` stores its data; must be
+  // located before `mProfileBuffer` so that it's destroyed after.
+  UniquePtr<BlocksRingBuffer> mBlocksRingBuffer;
+  UniquePtr<ProfileBuffer> mProfileBuffer;
 };
 
 }  // namespace baseprofiler